### PR TITLE
gcc uses bold to print function name in error messages

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: eaf179b8800e1323a2f8f6583156d93e4459b644).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88831062

File: src/util/cout_message.cpp [hunk 1]
```diff
@@ -157,7 +157,8 @@ void gcc_message_handlert::print(
       dest += command(1) + id2string(file) + ":" + command(0); // bold
     if(dest!="")
       dest+=' ';
-    dest+="In function '"+id2string(function)+"':\n";
+    dest += "In function " + command(1) + '\'' + id2string(function) + '\'' +
+            command(0) + ":\n";
```
Comment (tautschnig) [inline src/util/cout_message.cpp]: I believe this code should be rewritten so that it can use `bold` and `reset`.

**Review by tautschnig — CHANGES_REQUESTED** [review]

File: src/util/cout_message.cpp (hunk 1 above)
Comment (kroening) [reply src/util/cout_message.cpp]: done

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: a5e96f248e055b3b4b7a5d2587b54449f0129f5a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/89384568

File: src/util/cout_message.cpp [hunk 2]
```diff
@@ -175,11 +177,11 @@ void gcc_message_handlert::print(
       dest+=id2string(column)+": ";
 
     if(level==messaget::M_ERROR)
-      dest += command(31) + "error: "; // red
+      dest += string(messaget::red) + "error: ";
     else if(level==messaget::M_WARNING)
-      dest += command(95) + "warning: "; // bright magenta
+      dest += string(messaget::command(95)) + "warning: "; // bright magenta
```
Comment (peterschrammel) [inline src/util/cout_message.cpp]: Why not define this colour?

File: src/util/cout_message.cpp [hunk 3]
```diff
@@ -154,10 +154,12 @@ void gcc_message_handlert::print(
   if(!function.empty())
   {
     if(!file.empty())
-      dest += command(1) + id2string(file) + ":" + command(0); // bold
```
Comment (peterschrammel) [inline src/util/cout_message.cpp:158]: It would be nicer to refactor this to push into the stream.

**Review by peterschrammel — APPROVED** [review]

File: src/util/cout_message.cpp (hunk 2 above)
Comment (kroening) [reply src/util/cout_message.cpp]: done

File: src/util/cout_message.cpp (hunk 3 above)
Comment (kroening) [reply src/util/cout_message.cpp:158]: ok, added

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 99a705f4aa674d5925579970aa3d190a6aeece4f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/89507262
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

File: src/util/cout_message.cpp [hunk 4]
```diff
@@ -144,49 +144,51 @@ void gcc_message_handlert::print(
...
-    dest += string(messaget::bold);
+    const irep_idt file = location.get_file();
+    const irep_idt line = location.get_line();
+    const irep_idt column = location.get_column();
+    const irep_idt function = location.get_function();
+
+    if(!function.empty())
+    {
+      if(!file.empty())
+        out << string(messaget::bold) << id2string(file) << ':'
```
Comment (tautschnig) [inline src/util/cout_message.cpp]: The use of `string(...)` should no longer be necessary, and neither is `id2string(...)`

**Review by tautschnig — CHANGES_REQUESTED** [review]
What is the reason for "gcc_message_handlert: use stream directly"?

**Comment by kroening** [comment]
@tautschnig The direct usage of the stream was a request by @peterschrammel

File: src/util/cout_message.cpp (hunk 4 above)
Comment (kroening) [reply src/util/cout_message.cpp]: The id2string is removed.
Unfortunately, something like the string() is needed; the problem is that we are feeding into an std::ostream directly, an operator does not have access to the message handler.

**Review by tautschnig — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 1ea82edf1fdbb94b305997f7e8a341e3d6d7affb).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/89547479
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Pull request merged**